qiskit_experiments/data_processing/nodes.py

```diff
@@ -231,37 +223,19 @@ def _process(
             # level1 single mode
             dims = self._n_circs, self._n_shots, self._n_slots
 
-        processed_data = np.zeros(dims, dtype=float)
-        error_vals = np.zeros(dims, dtype=float)
+        projected_data = np.zeros(dims, dtype=object)
 
         for idx in range(self._n_slots):
             scale = self.scales[idx]
+            # error propagation is computed from data if any std error exists
             centered = np.array(
-                [datum[..., idx, iq] - self.means(qubit=idx, iq_index=iq) for iq in [0, 1]]
+                [data[..., idx, iq] - self.means(qubit=idx, iq_index=iq) for iq in [0, 1]]
             )
-            processed_data[..., idx] = (self._main_axes[idx] @ centered) / scale
-
-            if error is not None:
-                angle = np.arctan(self._main_axes[idx][1] / self._main_axes[idx][0])
-                error_vals[..., idx] = (
-                    np.sqrt(
-                        (error[..., idx, 0] * np.cos(angle)) ** 2
-                        + (error[..., idx, 1] * np.sin(angle)) ** 2
-                    )
-                    / scale
-                )
-
-        if self._n_circs == 1:
-            if error is None:
-                return processed_data[0], None
-            else:
-                return processed_data[0], error_vals[0]
+            projected_data[..., idx] = (self._main_axes[idx] @ centered) / scale
```

Comment:
This looks cool. Here we see that the uncertainties package nicely simplifies the error computation. Can we add a test to check correlated and anti-correlated IQ data? I think for this we can assuming the following setting:
* A SVD axis with a 45 degree angles.
* Anti-correlated IQ data: `[(-1, 1), (-0.5, 0.5), (0,0), (0.5, -0.5), (1, -1)]` with a 45 degree axis I would expect the standard error of the projected average point to be zero.
* Correlated IQ data: `[(-1, -1), (-0.5, -0.5), (0,0), (0.5, 0.5), (1, 1)]` with a 45 degree axis I would expect the standard error of the projected average point to be large.

Would such a test work? I think you can do this with a data processor that first averages the data to produce the average IQ point, in both cases the outcome should be `0+\-0.79` however the correlations between the avg(I) and avg(Q) should be different. Then use the SVD node, for the anti-correlated variables I expect `0+\-0` while for the positively correlated variables I expect `0+\-1.58`. I'm not sure if I got the number right but in any case I would expect to see a difference on the standard error but not on the mean. If this test works then I am fairly confident that this PR does what we want it to do.

Reply:
I tested with the code below
```python
svd_node = SVD()
svd_node._scales = [1.0]
svd_node._main_axes = [np.array([-1, -1])/np.sqrt(2)]
svd_node._means = [(0.0, 0.0A)]

processor = DataProcessor("memory", [AverageData(axis=1), svd_node])

res1 = processor({"memory": [[[-1, 1]], [[-0.5, 0.5]], [[0.0, 0.0]], [[0.5, -0.5]], [[1, -1]]]})
res2 = processor({"memory": [[[-1, -1]], [[-0.5, -0.5]], [[0.0, 0.0]], [[0.5, 0.5]], [[1, 1]]]})
```
however I cannot find the difference. The result is
```
# res1
array([-0.0+/-0.31622776601683794], dtype=object)

# res2
array([-0.0+/-0.31622776601683794], dtype=object)
```

If I drop the SVD node, then the output becomes
```
array([[0.0+/-0.31622776601683794, 0.0+/-0.31622776601683794]], dtype=object)
```
this is identical to the result `np.std([-1, -0.5, 0, 0.5, 1]) / np.sqrt(5)` so kind of expected. Probably we need to reconsider the average node as you mentioned in https://github.com/Qiskit/qiskit-experiments/pull/481#discussion_r749285013

Reply:
Or probably this is due to missing stdev in the training. Currently we ignore the stdev in SVD training because uncertainty doesn't support stdev method (I know how to implement uncertainties version of np.svd but I think this is overkill).

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/481/commits/504c3717165e585fc4ef487fa7ff8b4161eeb484